<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title>Introduction</title>

<script type="text/javascript">
window.onload = function() {
  var imgs = document.getElementsByTagName('img'), i, img;
  for (i = 0; i < imgs.length; i++) {
    img = imgs[i];
    // center an image if it is the only element of its parent
    if (img.parentElement.childElementCount === 1)
      img.parentElement.style.textAlign = 'center';
  }
};
</script>

<!-- Styles for R syntax highlighter -->
<style type="text/css">
   pre .operator,
   pre .paren {
     color: rgb(104, 118, 135)
   }

   pre .literal {
     color: #990073
   }

   pre .number {
     color: #099;
   }

   pre .comment {
     color: #998;
     font-style: italic
   }

   pre .keyword {
     color: #900;
     font-weight: bold
   }

   pre .identifier {
     color: rgb(0, 0, 0);
   }

   pre .string {
     color: #d14;
   }
</style>

<!-- R syntax highlighter -->
<script type="text/javascript">
var hljs=new function(){function m(p){return p.replace(/&/gm,"&amp;").replace(/</gm,"&lt;")}function f(r,q,p){return RegExp(q,"m"+(r.cI?"i":"")+(p?"g":""))}function b(r){for(var p=0;p<r.childNodes.length;p++){var q=r.childNodes[p];if(q.nodeName=="CODE"){return q}if(!(q.nodeType==3&&q.nodeValue.match(/\s+/))){break}}}function h(t,s){var p="";for(var r=0;r<t.childNodes.length;r++){if(t.childNodes[r].nodeType==3){var q=t.childNodes[r].nodeValue;if(s){q=q.replace(/\n/g,"")}p+=q}else{if(t.childNodes[r].nodeName=="BR"){p+="\n"}else{p+=h(t.childNodes[r])}}}if(/MSIE [678]/.test(navigator.userAgent)){p=p.replace(/\r/g,"\n")}return p}function a(s){var r=s.className.split(/\s+/);r=r.concat(s.parentNode.className.split(/\s+/));for(var q=0;q<r.length;q++){var p=r[q].replace(/^language-/,"");if(e[p]){return p}}}function c(q){var p=[];(function(s,t){for(var r=0;r<s.childNodes.length;r++){if(s.childNodes[r].nodeType==3){t+=s.childNodes[r].nodeValue.length}else{if(s.childNodes[r].nodeName=="BR"){t+=1}else{if(s.childNodes[r].nodeType==1){p.push({event:"start",offset:t,node:s.childNodes[r]});t=arguments.callee(s.childNodes[r],t);p.push({event:"stop",offset:t,node:s.childNodes[r]})}}}}return t})(q,0);return p}function k(y,w,x){var q=0;var z="";var s=[];function u(){if(y.length&&w.length){if(y[0].offset!=w[0].offset){return(y[0].offset<w[0].offset)?y:w}else{return w[0].event=="start"?y:w}}else{return y.length?y:w}}function t(D){var A="<"+D.nodeName.toLowerCase();for(var B=0;B<D.attributes.length;B++){var C=D.attributes[B];A+=" "+C.nodeName.toLowerCase();if(C.value!==undefined&&C.value!==false&&C.value!==null){A+='="'+m(C.value)+'"'}}return A+">"}while(y.length||w.length){var v=u().splice(0,1)[0];z+=m(x.substr(q,v.offset-q));q=v.offset;if(v.event=="start"){z+=t(v.node);s.push(v.node)}else{if(v.event=="stop"){var p,r=s.length;do{r--;p=s[r];z+=("</"+p.nodeName.toLowerCase()+">")}while(p!=v.node);s.splice(r,1);while(r<s.length){z+=t(s[r]);r++}}}}return z+m(x.substr(q))}function j(){function q(x,y,v){if(x.compiled){return}var u;var s=[];if(x.k){x.lR=f(y,x.l||hljs.IR,true);for(var w in x.k){if(!x.k.hasOwnProperty(w)){continue}if(x.k[w] instanceof Object){u=x.k[w]}else{u=x.k;w="keyword"}for(var r in u){if(!u.hasOwnProperty(r)){continue}x.k[r]=[w,u[r]];s.push(r)}}}if(!v){if(x.bWK){x.b="\\b("+s.join("|")+")\\s"}x.bR=f(y,x.b?x.b:"\\B|\\b");if(!x.e&&!x.eW){x.e="\\B|\\b"}if(x.e){x.eR=f(y,x.e)}}if(x.i){x.iR=f(y,x.i)}if(x.r===undefined){x.r=1}if(!x.c){x.c=[]}x.compiled=true;for(var t=0;t<x.c.length;t++){if(x.c[t]=="self"){x.c[t]=x}q(x.c[t],y,false)}if(x.starts){q(x.starts,y,false)}}for(var p in e){if(!e.hasOwnProperty(p)){continue}q(e[p].dM,e[p],true)}}function d(B,C){if(!j.called){j();j.called=true}function q(r,M){for(var L=0;L<M.c.length;L++){if((M.c[L].bR.exec(r)||[null])[0]==r){return M.c[L]}}}function v(L,r){if(D[L].e&&D[L].eR.test(r)){return 1}if(D[L].eW){var M=v(L-1,r);return M?M+1:0}return 0}function w(r,L){return L.i&&L.iR.test(r)}function K(N,O){var M=[];for(var L=0;L<N.c.length;L++){M.push(N.c[L].b)}var r=D.length-1;do{if(D[r].e){M.push(D[r].e)}r--}while(D[r+1].eW);if(N.i){M.push(N.i)}return f(O,M.join("|"),true)}function p(M,L){var N=D[D.length-1];if(!N.t){N.t=K(N,E)}N.t.lastIndex=L;var r=N.t.exec(M);return r?[M.substr(L,r.index-L),r[0],false]:[M.substr(L),"",true]}function z(N,r){var L=E.cI?r[0].toLowerCase():r[0];var M=N.k[L];if(M&&M instanceof Array){return M}return false}function F(L,P){L=m(L);if(!P.k){return L}var r="";var O=0;P.lR.lastIndex=0;var M=P.lR.exec(L);while(M){r+=L.substr(O,M.index-O);var N=z(P,M);if(N){x+=N[1];r+='<span class="'+N[0]+'">'+M[0]+"</span>"}else{r+=M[0]}O=P.lR.lastIndex;M=P.lR.exec(L)}return r+L.substr(O,L.length-O)}function J(L,M){if(M.sL&&e[M.sL]){var r=d(M.sL,L);x+=r.keyword_count;return r.value}else{return F(L,M)}}function I(M,r){var L=M.cN?'<span class="'+M.cN+'">':"";if(M.rB){y+=L;M.buffer=""}else{if(M.eB){y+=m(r)+L;M.buffer=""}else{y+=L;M.buffer=r}}D.push(M);A+=M.r}function G(N,M,Q){var R=D[D.length-1];if(Q){y+=J(R.buffer+N,R);return false}var P=q(M,R);if(P){y+=J(R.buffer+N,R);I(P,M);return P.rB}var L=v(D.length-1,M);if(L){var O=R.cN?"</span>":"";if(R.rE){y+=J(R.buffer+N,R)+O}else{if(R.eE){y+=J(R.buffer+N,R)+O+m(M)}else{y+=J(R.buffer+N+M,R)+O}}while(L>1){O=D[D.length-2].cN?"</span>":"";y+=O;L--;D.length--}var r=D[D.length-1];D.length--;D[D.length-1].buffer="";if(r.starts){I(r.starts,"")}return R.rE}if(w(M,R)){throw"Illegal"}}var E=e[B];var D=[E.dM];var A=0;var x=0;var y="";try{var s,u=0;E.dM.buffer="";do{s=p(C,u);var t=G(s[0],s[1],s[2]);u+=s[0].length;if(!t){u+=s[1].length}}while(!s[2]);if(D.length>1){throw"Illegal"}return{r:A,keyword_count:x,value:y}}catch(H){if(H=="Illegal"){return{r:0,keyword_count:0,value:m(C)}}else{throw H}}}function g(t){var p={keyword_count:0,r:0,value:m(t)};var r=p;for(var q in e){if(!e.hasOwnProperty(q)){continue}var s=d(q,t);s.language=q;if(s.keyword_count+s.r>r.keyword_count+r.r){r=s}if(s.keyword_count+s.r>p.keyword_count+p.r){r=p;p=s}}if(r.language){p.second_best=r}return p}function i(r,q,p){if(q){r=r.replace(/^((<[^>]+>|\t)+)/gm,function(t,w,v,u){return w.replace(/\t/g,q)})}if(p){r=r.replace(/\n/g,"<br>")}return r}function n(t,w,r){var x=h(t,r);var v=a(t);var y,s;if(v){y=d(v,x)}else{return}var q=c(t);if(q.length){s=document.createElement("pre");s.innerHTML=y.value;y.value=k(q,c(s),x)}y.value=i(y.value,w,r);var u=t.className;if(!u.match("(\\s|^)(language-)?"+v+"(\\s|$)")){u=u?(u+" "+v):v}if(/MSIE [678]/.test(navigator.userAgent)&&t.tagName=="CODE"&&t.parentNode.tagName=="PRE"){s=t.parentNode;var p=document.createElement("div");p.innerHTML="<pre><code>"+y.value+"</code></pre>";t=p.firstChild.firstChild;p.firstChild.cN=s.cN;s.parentNode.replaceChild(p.firstChild,s)}else{t.innerHTML=y.value}t.className=u;t.result={language:v,kw:y.keyword_count,re:y.r};if(y.second_best){t.second_best={language:y.second_best.language,kw:y.second_best.keyword_count,re:y.second_best.r}}}function o(){if(o.called){return}o.called=true;var r=document.getElementsByTagName("pre");for(var p=0;p<r.length;p++){var q=b(r[p]);if(q){n(q,hljs.tabReplace)}}}function l(){if(window.addEventListener){window.addEventListener("DOMContentLoaded",o,false);window.addEventListener("load",o,false)}else{if(window.attachEvent){window.attachEvent("onload",o)}else{window.onload=o}}}var e={};this.LANGUAGES=e;this.highlight=d;this.highlightAuto=g;this.fixMarkup=i;this.highlightBlock=n;this.initHighlighting=o;this.initHighlightingOnLoad=l;this.IR="[a-zA-Z][a-zA-Z0-9_]*";this.UIR="[a-zA-Z_][a-zA-Z0-9_]*";this.NR="\\b\\d+(\\.\\d+)?";this.CNR="\\b(0[xX][a-fA-F0-9]+|(\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)";this.BNR="\\b(0b[01]+)";this.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|\\.|-|-=|/|/=|:|;|<|<<|<<=|<=|=|==|===|>|>=|>>|>>=|>>>|>>>=|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~";this.ER="(?![\\s\\S])";this.BE={b:"\\\\.",r:0};this.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[this.BE],r:0};this.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[this.BE],r:0};this.CLCM={cN:"comment",b:"//",e:"$"};this.CBLCLM={cN:"comment",b:"/\\*",e:"\\*/"};this.HCM={cN:"comment",b:"#",e:"$"};this.NM={cN:"number",b:this.NR,r:0};this.CNM={cN:"number",b:this.CNR,r:0};this.BNM={cN:"number",b:this.BNR,r:0};this.inherit=function(r,s){var p={};for(var q in r){p[q]=r[q]}if(s){for(var q in s){p[q]=s[q]}}return p}}();hljs.LANGUAGES.cpp=function(){var a={keyword:{"false":1,"int":1,"float":1,"while":1,"private":1,"char":1,"catch":1,"export":1,virtual:1,operator:2,sizeof:2,dynamic_cast:2,typedef:2,const_cast:2,"const":1,struct:1,"for":1,static_cast:2,union:1,namespace:1,unsigned:1,"long":1,"throw":1,"volatile":2,"static":1,"protected":1,bool:1,template:1,mutable:1,"if":1,"public":1,friend:2,"do":1,"return":1,"goto":1,auto:1,"void":2,"enum":1,"else":1,"break":1,"new":1,extern:1,using:1,"true":1,"class":1,asm:1,"case":1,typeid:1,"short":1,reinterpret_cast:2,"default":1,"double":1,register:1,explicit:1,signed:1,typename:1,"try":1,"this":1,"switch":1,"continue":1,wchar_t:1,inline:1,"delete":1,alignof:1,char16_t:1,char32_t:1,constexpr:1,decltype:1,noexcept:1,nullptr:1,static_assert:1,thread_local:1,restrict:1,_Bool:1,complex:1},built_in:{std:1,string:1,cin:1,cout:1,cerr:1,clog:1,stringstream:1,istringstream:1,ostringstream:1,auto_ptr:1,deque:1,list:1,queue:1,stack:1,vector:1,map:1,set:1,bitset:1,multiset:1,multimap:1,unordered_set:1,unordered_map:1,unordered_multiset:1,unordered_multimap:1,array:1,shared_ptr:1}};return{dM:{k:a,i:"</",c:[hljs.CLCM,hljs.CBLCLM,hljs.QSM,{cN:"string",b:"'\\\\?.",e:"'",i:"."},{cN:"number",b:"\\b(\\d+(\\.\\d*)?|\\.\\d+)(u|U|l|L|ul|UL|f|F)"},hljs.CNM,{cN:"preprocessor",b:"#",e:"$"},{cN:"stl_container",b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:a,r:10,c:["self"]}]}}}();hljs.LANGUAGES.r={dM:{c:[hljs.HCM,{cN:"number",b:"\\b0[xX][0-9a-fA-F]+[Li]?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+(?:[eE][+\\-]?\\d*)?L\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+\\.(?!\\d)(?:i\\b)?",e:hljs.IMMEDIATE_RE,r:1},{cN:"number",b:"\\b\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"keyword",b:"(?:tryCatch|library|setGeneric|setGroupGeneric)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\.",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\d+(?![\\w.])",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\b(?:function)",e:hljs.IMMEDIATE_RE,r:2},{cN:"keyword",b:"(?:if|in|break|next|repeat|else|for|return|switch|while|try|stop|warning|require|attach|detach|source|setMethod|setClass)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"literal",b:"(?:NA|NA_integer_|NA_real_|NA_character_|NA_complex_)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"literal",b:"(?:NULL|TRUE|FALSE|T|F|Inf|NaN)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"identifier",b:"[a-zA-Z.][a-zA-Z0-9._]*\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"<\\-(?!\\s*\\d)",e:hljs.IMMEDIATE_RE,r:2},{cN:"operator",b:"\\->|<\\-",e:hljs.IMMEDIATE_RE,r:1},{cN:"operator",b:"%%|~",e:hljs.IMMEDIATE_RE},{cN:"operator",b:">=|<=|==|!=|\\|\\||&&|=|\\+|\\-|\\*|/|\\^|>|<|!|&|\\||\\$|:",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"%",e:"%",i:"\\n",r:1},{cN:"identifier",b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[hljs.BE],r:0},{cN:"string",b:"'",e:"'",c:[hljs.BE],r:0},{cN:"paren",b:"[[({\\])}]",e:hljs.IMMEDIATE_RE,r:0}]}};
hljs.initHighlightingOnLoad();
</script>



<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 13px;
}

body {
  max-width: 800px;
  margin: auto;
  padding: 1em;
  line-height: 20px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 {
   font-size:2.2em;
}

h2 {
   font-size:1.8em;
}

h3 {
   font-size:1.4em;
}

h4 {
   font-size:1.0em;
}

h5 {
   font-size:0.9em;
}

h6 {
   font-size:0.8em;
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre, img {
  max-width: 100%;
}
pre {
  overflow-x: auto;
}
pre code {
   display: block; padding: 0.5em;
}

code {
  font-size: 92%;
  border: 1px solid #ccc;
}

code[class] {
  background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * {
      background: transparent !important;
      color: black !important;
      filter:none !important;
      -ms-filter: none !important;
   }

   body {
      font-size:12pt;
      max-width:100%;
   }

   a, a:visited {
      text-decoration: underline;
   }

   hr {
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote {
      padding-right: 1em;
      page-break-inside: avoid;
   }

   tr, img {
      page-break-inside: avoid;
   }

   img {
      max-width: 100% !important;
   }

   @page :left {
      margin: 15mm 20mm 15mm 10mm;
   }

   @page :right {
      margin: 15mm 10mm 15mm 20mm;
   }

   p, h2, h3 {
      orphans: 3; widows: 3;
   }

   h2, h3 {
      page-break-after: avoid;
   }
}
</style>



</head>

<body>
<h2>Introduction</h2>

<p>MetaLonDA (METAgenomic LONgitudinal Differential Abundance method) is a method that identifies the significant time intervals of microbial features in longitudinal studies. MetaLonDA has the ability to handle the inconsistencies and common challenges associated with human studies, such as variable sample collection times and uneven number of time points along the subjectsâ€™ longitudinal study. The method employs a negative binomial distribution in conjunction with a semi-parametric SS-ANOVA to model the count reads. Then, it performs the significance testing based on unit time intervals using permutation testing procedure.</p>

<h2>Example</h2>

<p>An example dataset derived from a subset of stress-induced Yeast is available by running</p>

<pre><code class="r">library(MetaLonDA)

## Load read counts of 8 features from 100 samples. Samples are from 2 groups, 5 subjects per group, and 10 time points per subject.
data(metalonda_test_data)
#View(metalonda_test_data[,1:20])
</code></pre>

<pre><code class="r">## Create Group, Time, and ID annotation vectors
n.group = 2
n.sample = 5 
n.timepoints = 10
Group = factor(c(rep(&quot;A&quot;, n.sample*n.timepoints), rep(&quot;B&quot;,n.sample*n.timepoints)))
Time = rep(rep(1:n.timepoints, times = n.sample), 2)
ID = factor(rep(1:(2*n.sample), each = n.timepoints))

## Define the prediction timeponits 
points = seq(1, 10, length.out = 100)
</code></pre>

<h3>Test one feature</h3>

<pre><code class="r">## Create Group, Time, and ID annotation vectors
n.group = 2
n.sample = 5 
n.timepoints = 10
Group = factor(c(rep(&quot;A&quot;, n.sample*n.timepoints), rep(&quot;B&quot;,n.sample*n.timepoints)))
Time = rep(rep(1:n.timepoints, times = n.sample), 2)
ID = factor(rep(1:(2*n.sample), each = n.timepoints))

## Define the prediction timeponits 
points = seq(1, 10, length.out = 100)
</code></pre>

<p>In our example, we used 20 permutations just to showcase how MetaLonDA works. In real analysis, this number should be much higher.</p>

<h3>Test all features</h3>

<pre><code class="r">## Identify significant time intervals for all features: 
output.metalonda.all = metalondaAll(Count = metalonda_test_data, Time = Time, Group = Group,
  ID = ID, n.perm = 20, fit.method = &quot;nbinomial&quot;, num.intervals = 100, 
  parall = FALSE, pvalue.threshold = 0.05, adjust.method = &quot;BH&quot;, time.unit = &quot;hours&quot;, norm.method = &quot;none&quot;,
  prefix = &quot;Test&quot;)
</code></pre>

<pre><code>## Dimensionality check passed
## Prediction Points =   [1]  1.00  1.09  1.18  1.27  1.36  1.45  1.54  1.63  1.72  1.81  1.90
##  [12]  1.99  2.08  2.17  2.26  2.35  2.44  2.53  2.62  2.71  2.80  2.89
##  [23]  2.98  3.07  3.16  3.25  3.34  3.43  3.52  3.61  3.70  3.79  3.88
##  [34]  3.97  4.06  4.15  4.24  4.33  4.42  4.51  4.60  4.69  4.78  4.87
##  [45]  4.96  5.05  5.14  5.23  5.32  5.41  5.50  5.59  5.68  5.77  5.86
##  [56]  5.95  6.04  6.13  6.22  6.31  6.40  6.49  6.58  6.67  6.76  6.85
##  [67]  6.94  7.03  7.12  7.21  7.30  7.39  7.48  7.57  7.66  7.75  7.84
##  [78]  7.93  8.02  8.11  8.20  8.29  8.38  8.47  8.56  8.65  8.74  8.83
##  [89]  8.92  9.01  9.10  9.19  9.28  9.37  9.46  9.55  9.64  9.73  9.82
## [100]  9.91 10.00
## 
## Feature  =  OTU_1 
## Start MetaLonDA 
## Visualizing Feature =  OTU_1
</code></pre>

<pre><code>## Start Curve Fitting 
## Fitting: NB SS 
## Visualizing Splines of Feature =  OTU_1
</code></pre>

<pre><code>## Calculate Area Under the Fitted Curves 
## Start Permutation 
## # of Subjects =  10 
## Permutation # 1 
## Permutation # 2 
## Permutation # 3 
## Permutation # 4 
## Permutation # 5 
## Permutation # 6 
## Permutation # 7 
## Permutation # 8 
## Permutation # 9 
## Permutation # 10 
## Permutation # 11 
## Permutation # 12 
## Permutation # 13 
## Permutation # 14 
## Permutation # 15 
## Permutation # 16 
## Permutation # 17 
## Permutation # 18 
## Permutation # 19 
## Permutation # 20 
## p-value Adjustment Method =  BH 
## Visualizing Significant Intervals of Feature =  OTU_1
</code></pre>

<pre><code>## 
## 
## Feature  =  OTU_2 
## Start MetaLonDA 
## Visualizing Feature =  OTU_2
</code></pre>

<pre><code>## Start Curve Fitting 
## Fitting: NB SS 
## Visualizing Splines of Feature =  OTU_2
</code></pre>

<pre><code>## Calculate Area Under the Fitted Curves 
## Start Permutation 
## # of Subjects =  10 
## Permutation # 1 
## Permutation # 2 
## Permutation # 3 
## Permutation # 4 
## Permutation # 5 
## Permutation # 6 
## Permutation # 7 
## Permutation # 8 
## Permutation # 9 
## Permutation # 10 
## Permutation # 11 
## Permutation # 12 
## Permutation # 13 
## Permutation # 14 
## Permutation # 15 
## Permutation # 16 
## Permutation # 17 
## Permutation # 18 
## Permutation # 19 
## Permutation # 20 
## p-value Adjustment Method =  BH 
## Visualizing Significant Intervals of Feature =  OTU_2
</code></pre>

<pre><code>## 
## 
## Feature  =  OTU_3 
## Start MetaLonDA 
## Visualizing Feature =  OTU_3
</code></pre>

<pre><code>## Start Curve Fitting 
## Fitting: NB SS 
## Visualizing Splines of Feature =  OTU_3
</code></pre>

<pre><code>## Calculate Area Under the Fitted Curves 
## Start Permutation 
## # of Subjects =  10 
## Permutation # 1 
## Permutation # 2 
## Permutation # 3 
## Permutation # 4 
## Permutation # 5 
## Permutation # 6 
## Permutation # 7 
## Permutation # 8 
## Permutation # 9 
## Permutation # 10 
## Permutation # 11 
## Permutation # 12 
## Permutation # 13 
## Permutation # 14 
## Permutation # 15 
## Permutation # 16 
## Permutation # 17 
## Permutation # 18 
## Permutation # 19 
## Permutation # 20 
## p-value Adjustment Method =  BH 
## Visualizing Significant Intervals of Feature =  OTU_3
</code></pre>

<pre><code>## 
## 
## Feature  =  OTU_4 
## Start MetaLonDA 
## Visualizing Feature =  OTU_4
</code></pre>

<pre><code>## Start Curve Fitting 
## Fitting: NB SS 
## Visualizing Splines of Feature =  OTU_4
</code></pre>

<pre><code>## Calculate Area Under the Fitted Curves 
## Start Permutation 
## # of Subjects =  10 
## Permutation # 1 
## Permutation # 2 
## Permutation # 3 
## Permutation # 4 
## Permutation # 5 
## Permutation # 6 
## Permutation # 7 
## Permutation # 8 
## Permutation # 9 
## Permutation # 10 
## Permutation # 11 
## Permutation # 12 
## Permutation # 13 
## Permutation # 14 
## Permutation # 15 
## Permutation # 16 
## Permutation # 17 
## Permutation # 18 
## Permutation # 19 
## Permutation # 20 
## p-value Adjustment Method =  BH 
## Visualizing Significant Intervals of Feature =  OTU_4
</code></pre>

<pre><code>## 
## 
## Feature  =  OTU_5 
## Start MetaLonDA 
## Visualizing Feature =  OTU_5
</code></pre>

<pre><code>## Start Curve Fitting 
## Fitting: NB SS 
## Visualizing Splines of Feature =  OTU_5
</code></pre>

<pre><code>## Calculate Area Under the Fitted Curves 
## Start Permutation 
## # of Subjects =  10 
## Permutation # 1 
## Permutation # 2 
## Permutation # 3 
## Permutation # 4 
## Permutation # 5 
## Permutation # 6 
## Permutation # 7 
## Permutation # 8 
## Permutation # 9 
## Permutation # 10 
## Permutation # 11 
## Permutation # 12 
## Permutation # 13 
## Permutation # 14 
## Permutation # 15 
## Permutation # 16 
## Permutation # 17 
## Permutation # 18 
## Permutation # 19 
## Permutation # 20 
## p-value Adjustment Method =  BH 
## Visualizing Significant Intervals of Feature =  OTU_5
</code></pre>

<pre><code>## 
## 
## Feature  =  OTU_6 
## Start MetaLonDA 
## Visualizing Feature =  OTU_6
</code></pre>

<pre><code>## Start Curve Fitting 
## Fitting: NB SS 
## Visualizing Splines of Feature =  OTU_6
</code></pre>

<pre><code>## Calculate Area Under the Fitted Curves 
## Start Permutation 
## # of Subjects =  10 
## Permutation # 1 
## Permutation # 2 
## Permutation # 3 
## Permutation # 4 
## Permutation # 5 
## Permutation # 6 
## Permutation # 7 
## Permutation # 8 
## Permutation # 9 
## Permutation # 10 
## Permutation # 11 
## Permutation # 12 
## Permutation # 13 
## Permutation # 14 
## Permutation # 15 
## Permutation # 16 
## Permutation # 17 
## Permutation # 18 
## Permutation # 19 
## Permutation # 20 
## p-value Adjustment Method =  BH 
## Visualizing Significant Intervals of Feature =  OTU_6
</code></pre>

<pre><code>## 
## 
## Feature  =  OTU_7 
## Start MetaLonDA 
## Visualizing Feature =  OTU_7
</code></pre>

<pre><code>## Start Curve Fitting 
## Fitting: NB SS 
## Visualizing Splines of Feature =  OTU_7
</code></pre>

<pre><code>## Calculate Area Under the Fitted Curves 
## Start Permutation 
## # of Subjects =  10 
## Permutation # 1 
## Permutation # 2 
## Permutation # 3 
## Permutation # 4 
## Permutation # 5 
## Permutation # 6 
## Permutation # 7 
## Permutation # 8 
## Permutation # 9 
## Permutation # 10 
## Permutation # 11 
## Permutation # 12 
## Permutation # 13 
## Permutation # 14 
## Permutation # 15 
## Permutation # 16 
## Permutation # 17 
## Permutation # 18 
## Permutation # 19 
## Permutation # 20 
## p-value Adjustment Method =  BH 
## No Significant Intevals Found 
## 
## 
## Feature  =  OTU_8 
## Start MetaLonDA 
## Visualizing Feature =  OTU_8
</code></pre>

<pre><code>## Start Curve Fitting 
## Fitting: NB SS 
## Visualizing Splines of Feature =  OTU_8
</code></pre>

<pre><code>## Calculate Area Under the Fitted Curves 
## Start Permutation 
## # of Subjects =  10 
## Permutation # 1 
## Permutation # 2 
## Permutation # 3 
## Permutation # 4 
## Permutation # 5 
## Permutation # 6 
## Permutation # 7 
## Permutation # 8 
## Permutation # 9 
## Permutation # 10 
## Permutation # 11 
## Permutation # 12 
## Permutation # 13 
## Permutation # 14 
## Permutation # 15 
## Permutation # 16 
## Permutation # 17 
## Permutation # 18 
## Permutation # 19 
## Permutation # 20 
## p-value Adjustment Method =  BH 
## No Significant Intevals Found
</code></pre>

<h2>Session information</h2>

<pre><code class="r">sessionInfo()
</code></pre>

<pre><code>## R version 3.3.2 (2016-10-31)
## Platform: x86_64-apple-darwin13.4.0 (64-bit)
## Running under: macOS Sierra 10.12.4
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
## [1] rmarkdown_1.6     MetaLonDA_1.1.0   doParallel_1.0.11 iterators_1.0.9  
## [5] foreach_1.4.4     caTools_1.17.1    plyr_1.8.4        gss_2.1-7        
## [9] ggplot2_2.2.1    
## 
## loaded via a namespace (and not attached):
##  [1] Biobase_2.34.0             edgeR_3.16.5              
##  [3] splines_3.3.2              gtools_3.5.0              
##  [5] assertthat_0.1             Formula_1.2-1             
##  [7] stats4_3.3.2               latticeExtra_0.6-28       
##  [9] yaml_2.1.14                RSQLite_1.1-2             
## [11] backports_1.0.5            lattice_0.20-34           
## [13] limma_3.30.11              digest_0.6.12             
## [15] GenomicRanges_1.26.3       RColorBrewer_1.1-2        
## [17] XVector_0.14.0             checkmate_1.8.2           
## [19] colorspace_1.3-2           htmltools_0.3.5           
## [21] Matrix_1.2-7.1             DESeq2_1.14.1             
## [23] XML_3.98-1.5               devtools_1.12.0           
## [25] genefilter_1.56.0          zlibbioc_1.20.0           
## [27] xtable_1.8-2               scales_0.5.0              
## [29] gdata_2.17.0               BiocParallel_1.8.1        
## [31] git2r_0.18.0               htmlTable_1.9             
## [33] tibble_1.3.3               annotate_1.52.1           
## [35] IRanges_2.8.1              withr_1.0.2               
## [37] SummarizedExperiment_1.4.0 nnet_7.3-12               
## [39] BiocGenerics_0.20.0        lazyeval_0.2.0            
## [41] crayon_1.3.2               survival_2.40-1           
## [43] magrittr_1.5               evaluate_0.10             
## [45] memoise_1.0.0              gplots_3.0.1              
## [47] xml2_1.1.1                 foreign_0.8-67            
## [49] BiocInstaller_1.24.0       tools_3.3.2               
## [51] data.table_1.10.0          matrixStats_0.51.0        
## [53] stringr_1.2.0              S4Vectors_0.12.1          
## [55] locfit_1.5-9.1             munsell_0.4.3             
## [57] glmnet_2.0-5               cluster_2.0.5             
## [59] rversions_1.0.3            AnnotationDbi_1.36.2      
## [61] GenomeInfoDb_1.10.3        rlang_0.1.1               
## [63] grid_3.3.2                 RCurl_1.95-4.8            
## [65] rstudioapi_0.6             htmlwidgets_0.8           
## [67] bitops_1.0-6               base64enc_0.1-3           
## [69] labeling_0.3               metagenomeSeq_1.16.0      
## [71] gtable_0.2.0               codetools_0.2-15          
## [73] curl_3.1                   DBI_0.5-1                 
## [75] roxygen2_6.0.1             R6_2.2.2                  
## [77] gridExtra_2.2.1            knitr_1.15.1              
## [79] rprojroot_1.2              commonmark_1.2            
## [81] Hmisc_4.0-2                desc_1.1.0                
## [83] KernSmooth_2.23-15         stringi_1.1.2             
## [85] Rcpp_0.12.14               geneplotter_1.52.0        
## [87] rpart_4.1-10               acepack_1.4.1
</code></pre>

<h2>References</h2>

<p>Metwally, Ahmed A., Jie Yang, Christian Ascoli, Yang Dai, Patricia W. Finn, and David L. Perkins. &ldquo;MetaLonDA: a flexible R package for identifying time intervals of differentially abundant features in metagenomic longitudinal studies&rdquo;, <em>Microbiome</em>, 2018.</p>

<p>Metwally, Ahmed A., Patricia W. Finn, Yang Dai, and David L. Perkins. &ldquo;Detection of Differential Abundance Intervals in Longitudinal Metagenomic Data Using Negative Binomial Smoothing Spline ANOVA.&rdquo; <em>ACM BCB</em>, 2017. </p>

<h2>Bugs and Suggestions</h2>

<p>MetaLonDA is under active research development. Please report any bugs/suggestions to Ahmed Metwally (<a href="mailto:ametwa2@uic.edu">ametwa2@uic.edu</a>).</p>

</body>

</html>
